src/main/java/trackr/storage/JsonAdaptedMenuItem.java

```diff
@@ -0,0 +1,85 @@
+package trackr.storage;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import trackr.commons.exceptions.IllegalValueException;
+import trackr.model.commons.Name;
+import trackr.model.menu.ItemCost;
+import trackr.model.menu.ItemName;
+import trackr.model.menu.ItemPrice;
+import trackr.model.menu.MenuItem;
+
+/**
+ * Jackson-friendly version of {@link MenuItem}.
+ */
+class JsonAdaptedMenuItem {
+
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Menu Item's %s field is missing!";
+
+    private final String itemName;
+    private final String itemCost;
+    private final String itemPrice;
+    private final String itemProfit;
+
+    /**
+     * Constructs a {@code JsonAdaptedMenuItem} with the given menu item details.
+     */
+    @JsonCreator
+    public JsonAdaptedMenuItem(@JsonProperty("itemName") String itemName,
+                               @JsonProperty("itemCost") String itemCost,
+                               @JsonProperty("itemPrice") String itemPrice,
+                               @JsonProperty("itemProfit") String itemProfit) {
+        this.itemName = itemName;
+        this.itemCost = itemCost;
+        this.itemPrice = itemPrice;
+        this.itemProfit = itemProfit;
+    }
+
+    /**
+     * Converts a given {@code MenuItem} into this class for Jackson use.
+     */
+    public JsonAdaptedMenuItem(MenuItem source) {
+        itemName = source.getItemName().getName();
+        itemCost = source.getItemCost().toJsonString();
+        itemPrice = source.getItemPrice().toJsonString();
```

Comment:
String of `float` here

Reply:
Thank you, I will send another PR after merging this PR.